### PR TITLE
feat: post page/modal widget to set bookmark reminder

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- New bookmark icon.
- After bookmarking the post, we will display a widget to select the preferred reminder
- Refactored and introduced a `useJustBookmarked` hook that is reused in some places.
- Introduced `PostReminderOptions` that can be reused on the feed card.

Preview:

https://github.com/dailydotdev/apps/assets/13744167/e324ac09-e634-477b-8c4f-9248c0ff76a2

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

MI-433 #done

### Preview domain
https://mi-433.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/modals/common.tsx [hunk 1]
```diff
@@ -162,6 +162,7 @@ export const modals = {
   [LazyModal.SubmitSquadForReview]: SubmitSquadForReviewModal,
   [LazyModal.PrivilegedMembers]: PrivilegedMemberModal,
   [LazyModal.FirefoxPrivacy]: FirefoxPrivacyModal,
+  [LazyModal.ReadItLater]: (): void => null,
```
Comment (rebelchris) [inline packages/shared/src/components/modals/common.tsx:165]: TODO: Blocked my Luca's PR I assume

File: packages/shared/src/hooks/notifications/useBookmarkReminderEnrollment.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useJustBookmarked } from '../bookmark';
+import { useConditionalFeature } from '../useConditionalFeature';
+import { feature } from '../../lib/featureManagement';
+import { Post } from '../../graphql/posts';
+
+export const useBookmarkReminderEnrollment = (post: Post): boolean => {
```
Comment (rebelchris) [inline packages/shared/src/hooks/notifications/useBookmarkReminderEnrollment.ts:6]: Oeh nice, love this idea also easier cleanup I guess 😅 (as we'll always need the justBookmarked part)

File: packages/shared/src/hooks/useBookmarkProvider.ts [hunk 3]
```diff
@@ -1,25 +1,20 @@
...
-  bookmarked?: boolean;
-}
-
 const useBookmarkProvider = ({
   bookmarked = false,
 }: UseBookmarkProviderProps): UseBookmarkProviderReturn => {
-  const isBookmarked = useRef<boolean>(bookmarked);
   const { feedName } = useActiveFeedNameContext();
-  const justBookmarked = bookmarked !== isBookmarked.current && bookmarked;
+  const { justBookmarked, wasBookmarked } = useJustBookmarked({ bookmarked });
```
Comment (rebelchris) [inline packages/shared/src/hooks/useBookmarkProvider.ts:13]: This still supports the old one (highlighting old bookmarks) vs now also showing the new popup I guess?

**Review by rebelchris — APPROVED** [review]
We just have to carefully monitor the old behaviour vs the new one (that both work) mainly because the old one is hard to test 😅

File: packages/shared/src/hooks/useBookmarkProvider.ts (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/hooks/useBookmarkProvider.ts:13]: Yes, should still support as intended. Just changed the variable name from `isBookmarked` to `wasBookmarked` - to be clearer.
Screenshot 2024-07-11 at 9 46 23 PM

**Review by nensidosari — APPROVED** [review]
Great work!

File: packages/shared/src/components/modals/common.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/modals/common.tsx:165]: Yeah. Luca's PR is about to be merged. I think I will wait for that before I merge this one and have this part updated.

[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/modals/common.tsx]: Updated 🥳

**Pull request merged**